v1.1/parallel-statement-execution.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Parallel Statement Execution
+summary: The parallel statement execution feature allows parallel execution of multiple independent SQL statements within a transaction.
+toc: false
+---
+
+CockroachDB supports parallel execution of independent [`INSERT`](insert.html), [`UPDATE`](update.html), [`UPSERT`](upsert.html), and [`DELETE`](delete.html) statements within a single transaction. Parallel execution of SQL statements reduces the overall execution time of multiple SQL statements.
+
+<div id="toc"></div>
+
+## Why Use Parallel Statement Execution
+
+Normally, applications issue multiple SQL statements within a single transaction and SQL engines execute each statement sequentially. Because the statements are executed locally, the results can be delivered fast. But CockroachDB being a distributed database, execution of SQL statements takes longer than normal SQL engines because of high round-trip latencies required for consensus. To counteract the issue and optimize performance, CockroachDB supports parallel execution of SQL statements. Thus, despite the higher latency of single data-modifying statements, the overall execution latency of multiple statements can be reduced.
+
+## How Parallel Statement Execution Works
+
+SQL statements within a single transaction can be executed in parallel if the statements are independent. CockroachDB considers SQL statements within a single transaction to be independent if they operate on independent tables and their execution can be safely reordered without affecting their results. 
+
+Normal execution of SQL statements returns a successful return value immediately from the server. After the return value is received, the next SQL statement is executed. Appending the `RETURNING NOTHING` clause with SQL statements prevents the server from sending return values and allows CockroachDB to execute the statements in parallel. Thus to execute statements in parallel, append the `RETURNING NOTHING` suffix to the SQL statements. 
```

Comment:
It doesn't have to be "successful", it can also return an error.

For reference, the equivalent paragraph in my blog post was:
> The challenge with introducing any form of parallelism in SQL is that the language is traditionally exposed as a statement at-a-time conversational API. This means that a client sends a statement to a SQL database, the database processes the statement, and finally the statement’s results are sent back to the client. Only at this time can the next statement be issued. In order to gain any parallelism between these statements, we first needed to break this mold.

> We did this by introducing a new syntax called `RETURNING NOTHING`. Usually statements like INSERT and UPDATE return the number of rows affected. However, they can be augmented with a `RETURNING` clause to specify a set of columns from the affected rows to return. Much like other `RETURNING` clauses, our new `RETURNING NOTHING` clause is appended on to the end of statements like INSERT and UPDATE. Instead of returning a set of columns though, this new syntax does exactly what it says; it returns nothing. In doing so, it creates an opportunity for CockroachDB to return to the client with nothing but an acknowledgement and allow it to issue more statements before the first one has finished executing. When a second statement is issued, it can then be run in parallel with the first statement. Furthermore, if a chain of statements within a transaction are run with the `RETURNING NOTHING` syntax, they can all be run together and in parallel!